jinahub/simple/__init__.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Tuple, Optional, List
+
+import numpy as np
+from jina import Executor, DocumentArray, requests, Document
+from jina.types.arrays.memmap import DocumentArrayMemmap
+
+
+class SimpleIndexer(Executor):
+    """A simple indexer that stores all the Document data together, in a DocumentArrayMemmap object
+
+    To be used as a unified indexer, combining both indexing and searching"""
+
+    def __init__(self, index_file_name: str,
+                 default_traversal_paths: Optional[List[str]] = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self._docs = DocumentArrayMemmap(self.workspace + f'/{index_file_name}')
+        self.default_traversal_paths = default_traversal_paths or ['r']
+
+    @requests(on='/index')
+    def index(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+        """All Documents to the DocumentArray
+        :param docs: the docs to add
+        :param parameters: the parameters dictionary
+        """
+        traversal_path = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+        flat_docs = docs.traverse_flat(traversal_path)
+        self._docs.extend(flat_docs)
+
+    @requests(on='/search')
+    def search(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+        """Perform a vector similarity search and retrieve the full Document match
+
+        :param docs: the Documents to search with
+        :param parameters: the parameters for the search"""
+        traversal_path = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+        flat_docs = docs.traverse_flat(traversal_path)
+        a = np.stack(flat_docs.get_attributes('embedding'))
+        b = np.stack(self._docs.get_attributes('embedding'))
```

Comment:
Instead of generating `b` for every search, we can add a `self._flush` flag to denote whether there is no document added.
```suggestion
        b = np.stack(self._docs.get_attributes('embedding'))
```

Reply:
Like so? https://github.com/jina-ai/executor-indexers/pull/56/files#diff-9abd8df8a48627b91a073faa54b285dc648a66b055d36b9a0821a366e51d3597R46